File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/screens/pages/FragmentListView.java
package ru.usedesk.knowledgebase_gui.screens.pages;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import ru.usedesk.knowledgebase_gui.R;
import ru.usedesk.knowledgebase_gui.screens.common.DataViewModel;
import ru.usedesk.knowledgebase_gui.screens.common.FragmentDataView;

public abstract class FragmentListView<V, T extends DataViewModel<List<V>>>
        extends FragmentDataView<List<V>, T> {

    private RecyclerView recyclerViewSections;

    public FragmentListView() {
        super(R.layout.usedesk_fragment_list);
    }

    @Override
    protected void onView(@NonNull View view) {
        super.onView(view);
        recyclerViewSections = view.findViewById(R.id.rv_list);
    }

    @Override
    protected void setDataView(List<V> data) {
        recyclerViewSections.setAdapter(getAdapter(data));
        recyclerViewSections.setLayoutManager(new LinearLayoutManager(getContext()));
    }

    protected abstract RecyclerView.Adapter getAdapter(List<V> list);
}

File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/pages/FragmentListView.kt
package ru.usedesk.knowledgebase_gui.pages

import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import ru.usedesk.common_gui.UsedeskBinding
import ru.usedesk.knowledgebase_gui.R
import ru.usedesk.knowledgebase_gui.common.FragmentDataView

internal abstract class FragmentListView<DATA, BINDING : UsedeskBinding>(
        layoutId: Int,
        styleId: Int
) : FragmentDataView<List<DATA>, BINDING>(layoutId, styleId) {
    private lateinit var recyclerViewSections: RecyclerView

    override fun onView(view: View) {
        super.onView(view)
        recyclerViewSections = view.findViewById(R.id.rv_list)
    }

    override fun setDataView(data: List<DATA>) {
        recyclerViewSections.adapter = getAdapter(data)
        recyclerViewSections.layoutManager = LinearLayoutManager(context)
    }

    protected abstract fun getAdapter(list: List<DATA>): RecyclerView.Adapter<*>
}